### ISSUE TITLE
Support .mdx files

### ISSUE THREAD

**Issue by bencmbrook** [remarkjs/vscode-remark#129]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

vscode-remark only recognizes markdown files https://github.com/remarkjs/vscode-remark/blob/a6b066c1c3df19e59de4de49152c878d8104264a/src/extension.js#L17

but there is now a dedicated MDX language in VS Code. it would be awesome if we could use remark lint on our MDX files!

### Solution

Update the documentSelector to include MDX files

### Alternatives

You can force VS Code to treat MDX as Markdown, which allows vscode-remark to work, but it comes at the cost of all the other language features of MDX in VS Code via [vscode-mdx](https://github.com/mdx-js/mdx-analyzer/tree/main/packages/vscode-mdx)

```
  "files.associations": {
    "*.mdx": "markdown",
  },
```

**Comment by JounQin** [comment]
MDX and markdown are definitely different, and there is already `vscode-mdx`.

**Comment by remcohaszing** [comment]
`vscode-mdx` provides MDX language support. I would also like to see `unified-engine` based remark support for MDX.

Related issues:
- https://github.com/unifiedjs/unified-engine/issues/56
- https://github.com/mdx-js/mdx/issues/2378

I think those issues need to be resolved first. Then we can investigate support in `remark-language-server` / `vscode-remark`. We can implement something like probing in [`vscode-eslint`](https://github.com/microsoft/vscode-eslint).

**Comment by bencmbrook** [comment]
Thank you! Yes my use-case is aligned to that @remcohaszing — showing our remark lint issues in MDX files with IntelliSense. 

Our workaround works today for linting MDX with vscode-remark (by using [remark-mdx](https://github.com/mdx-js/mdx/tree/main/packages/remark-mdx) in our `.remarkrc` file and forcing VS Code to treat MDX as Markdown via `"files.associations"`) but it comes at the tradeoff of the vscode-mdx extension no longer recognizing the file, since it's treated as Markdown.

**Comment by bencmbrook** [comment]
@remcohaszing given the extension works with MDX in our workaround... in theory, would updating this to simply include MDX files work?
https://github.com/remarkjs/vscode-remark/blob/a6b066c1c3df19e59de4de49152c878d8104264a/src/extension.js#L17
Ideally there would be an exposed setting as well that toggles that so it's opt-in. Also understand if you prefer to go with the more foundational work of handling rule overrides.

**Comment by Ayc0** [comment]
Cf https://github.com/orgs/mdx-js/discussions/2067#discussioncomment-3023340

**Comment by bencmbrook** [comment]
> Cf https://github.com/orgs/mdx-js/discussions/2067#discussioncomment-3023340
> 
> Remark-cli and thus remark-vscode are the recommended tools to format MDX

Yeah I'm using that for CLI formatting today, but this issue is primarily about highlighting lint errors in the same way that Prettier would, like so:

<img width="643" alt="Screenshot 2024-06-17 at 10 37 15 AM" src="https://github.com/remarkjs/vscode-remark/assets/7354176/3dd64634-6aae-4ef9-a521-a7e9567dad4b">

This is forcebly enabled in my VSCode via the method described in the **Alternatives** section in the issue. And these rules are specified via `.remarkrc`

**Comment by tjx666** [comment]
can't find a way to use remark to format .mdx now

**Comment by JounQin** [comment]
> can't find a way to use remark to format .mdx now

Or raise a PR fox fixing this.

By the way, programmably using `remark-mdx` + `remark-stringify` should just work AFAIK, we'using it perfectly at https://app.unpkg.com/@alauda/doom@0.20.0/files/lib/plugins/replace/remark-replace.js#L138-141
